agave/models/mongo/mongo_model.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Optional
+
+from cuenca_validations.typing import DictStrAny
+from mongoengine import Document, DoesNotExist, Q
+
+from agave.exc import ObjectDoesNotExist
+from agave.lib.mongoengine.model_helpers import mongo_to_dict
+from agave.models.base import BaseModel
+
+
+class MongoModel(BaseModel, Document):
+    meta = {'allow_inheritance': True}
+
+    def dict(self) -> DictStrAny:
+        return self._dict(mongo_to_dict)
+
+    @classmethod
+    def retrieve(cls, id: str, user_id: Optional[str] = None):
```

Comment:
refer to https://www.python.org/dev/peps/pep-0570/

should be
```python
def retrieve(cls, id: str, *, user_id: Optional[str] = None):
```

This means that it, if `user_id` is passed in`, is _must_ be passed in as a kwarg. `retrieve('222', '3333')` will be rejected and only `retrieve('222', user_id='3333')` will be accepted. This forces greater clarity in the code and allows more flexibility if new kwargs are added in the future.

Reply:
Done!